src/codecoverage/backend/tests/test_coverage.py

```diff
@@ -28,6 +31,60 @@ def test_coverage_supported_extensions_api(client):
     assert data['previous_rev'] == '5401938bde37d0e7f1016bbd7694e72bdbf5e9a1'
 
 
+def mock_coverage_by_changeset_job_success(job_changeset):
+    from tests.conftest import coverage_builds as get_coverage_builds
+    coverage_builds = get_coverage_builds()
+    for changeset, expected in coverage_builds['info'].items():
+        if changeset == job_changeset or changeset[:12] == job_changeset[:12]:
+            return expected
+    raise NotImplementedError('Not implemented return values for changeset %s' % job_changeset)
+
+
+def test_coverage_by_changeset(coverage_builds):
+    from rq import Queue
+    from codecoverage_backend import api
+
+    # patch the queue to be sync to allow it run without workers. http://python-rq.org/docs/testing/
+    with mock.patch('codecoverage_backend.api.q', Queue(connection=FakeStrictRedis(singleton=False))) as q:
+        # patch the mock_coverage_by_changeset
+        with mock.patch('codecoverage_backend.api.coverage_by_changeset_job', mock_coverage_by_changeset_job_success):
+            assert q.is_empty()
+
+            # Get changeset coverage information
+            for changeset, expected in coverage_builds['info'].items():
+                result, code = api.coverage_by_changeset(changeset)
+                assert code == 202
+
+            # test that in the case of exception it will return 500
+            result, code = api.coverage_by_changeset('mozilla test changeset')
+            assert code == 202
+
+            # run simple worker to run all tasks
+            w = SimpleWorker([q], connection=q.connection)
+            w.work(burst=True)
+
+            # Everything should be 200 now
+            for changeset, expected in coverage_builds['info'].items():
+                result, code = api.coverage_by_changeset(changeset)
+                assert code == 200
+
+            # except the incorrect changeset, should be 500
+            result, code = api.coverage_by_changeset('mozilla test changeset')
+            assert code == 500
+
+
+def test_coverage_summary_by_changeset(coverage_builds):
+    from rq import Queue
+    from codecoverage_backend import api
+
+    # patch the queue to be sync to allow it run without workers. http://python-rq.org/docs/testing/
```

Comment:
Could you do the same you did for `test_coverage_by_changeset` here? With the tests for 200 and 500

Reply:
ah I see, sorry I forgot about it